provisioning/miniprov/hare_mp/validate.py

```diff
@@ -0,0 +1,43 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Setup utility for Hare to configure Hare related settings, e.g. logrotate,
+# report unsupported features, etc.
+
+from hare_mp.store import ConfStoreProvider
+from hare_mp.store import ValueProvider
+
+class Validate:
+    def __init__(self, provider: ValueProvider):
+        super().__init__()
+        self.provider = provider
+
+    def is_first_node_in_storage_set(self) -> bool:
+        machine_id = self.provider.get_machine_id()
```

Comment:
[optional] If Validator then invokes `self.provider.get()` anyway, does it really make sense to encapsulate two other keys into `get_machine()` and `get_cluster_id()`?

Reply:
It feels like it makes more sense to make `get_machine()` and `get_cluster_id()` be the part of this particular class instead of ValueProvider.

Reply:
>If Validator then invokes self.provider.get() anyway, does it really make sense to encapsulate two other keys into get_machine() and get_cluster_id()?
Idea is to keep all the conf-store related functions at one place, close to conf-store. Yes, we are still invoking provider but imagine get_cluster_id() and get_machine_id() as helpers. I expect we may need to access at least these 2 KVs frequently as most of the keys need them. Thus keeping a single function would help us to make future changes to the keys as they will be in one place instead of getting spread across multiple places.

I think having them in ValueProvider keeps all the conf-store related keys and functionality at one place.

Reply:
Ok, fine